ml-agents-envs/mlagents/envs/tests/test_subprocess_env_manager.py

```diff
@@ -0,0 +1,107 @@
+import unittest.mock as mock
+from unittest.mock import Mock, MagicMock
+import unittest
+import cloudpickle
+from mlagents.envs.subprocess_env_manager import StepInfo
+
+from mlagents.envs.subprocess_env_manager import (
+    SubprocessEnvManager,
+    EnvironmentResponse,
+    EnvironmentCommand,
+    worker,
+)
+from mlagents.envs.base_unity_environment import BaseUnityEnvironment
+
+
+def mock_env_factory(worker_id: int):
+    return mock.create_autospec(spec=BaseUnityEnvironment)
+
+
+class MockEnvWorker:
+    def __init__(self, worker_id, resp=None):
+        self.worker_id = worker_id
+        self.process = None
+        self.conn = None
+        self.send = Mock()
+        self.recv = Mock(return_value=resp)
+
+
+class SubprocessEnvManagerTest(unittest.TestCase):
+    def test_environments_are_created(self):
+        SubprocessEnvManager.create_worker = MagicMock()
+        env = SubprocessEnvManager(mock_env_factory, 2)
+        # Creates two processes
+        self.assertEqual(
```

Comment:
nit: can replace this with `env.create_worker.assert_has_calls([mock.call(0, mock_env_factory), mock.call(1, mock_env_factory)])`